python/pyspark/errors/exceptions/base.py

```diff
@@ -31,6 +32,7 @@ def __init__(
         message: Optional[str] = None,
         error_class: Optional[str] = None,
         message_parameters: Optional[Dict[str, str]] = None,
+        data: Optional[List[Row]] = None,
```

Comment:
Is this typing correct? seems we append `diff_rows.append((r1, r2))`?

Reply:
On second thought, using `Iterable` seems more reasonable in this case. Let me update it again.

Reply:
BTW, just realized that `mypy` doesn't throw an error when we pass a `list` to the `Tuple` or a `tuple` to the `List`. Just FYI 🤔